projects/components/src/file-upload/file-upload.component.ts

```diff
@@ -0,0 +1,133 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, EventEmitter, Input, Output } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { IconType } from '@hypertrace/assets-library';
+import { Color } from '@hypertrace/common';
+import { isNil } from 'lodash-es';
+import { FileItem } from '../file-display/file-item';
+import { IconSize } from '../icon/icon-size';
+
+@Component({
+  selector: 'ht-file-upload',
+  styleUrls: ['./file-upload.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      multi: true,
+      useExisting: FileUploadComponent
+    }
+  ],
+  template: `
+    <div class="file-upload">
+      <div
+        class="upload-section"
+        [ngClass]="{ 'drag-over': this.isDragOver, disabled: this.disabled }"
+        htDropZone
+        (dragOver)="this.onDragOver($event)"
+        (dropped)="this.onDrop($event)"
+      >
+        <ht-icon
+          class="cloud-upload-icon"
+          icon="${IconType.CloudUpload}"
+          size="${IconSize.ExtraLarger}"
+          color="${Color.Blue4}"
+        ></ht-icon>
+        <input type="file" multiple="multiple" (change)="this.onFilesSelection($event)" hidden #fileInput />
+        <div class="title">
+          <div class="click-to-upload" (click)="fileInput.click()">Click to upload</div>
+          or drag and drop
+        </div>
+        <div class="sub-text">{{ this.subText }}</div>
+      </div>
+      <div class="files-section">
+        <ht-file-display
+          *ngFor="let file of this.files; let index = index"
+          [file]="file"
+          (deleteClick)="this.deleteFile(index)"
+        >
+        </ht-file-display>
+      </div>
+    </div>
+  `
+})
+export class FileUploadComponent implements ControlValueAccessor {
+  @Input()
+  public files: FileItem[] = []; // Can be used to update progress
+
+  @Input()
+  public subText: string = 'max size 2 GB';
+
+  @Input()
+  public disabled: boolean = false;
+
+  @Output()
+  private readonly fileUpload: EventEmitter<FileItem[]> = new EventEmitter();
+
+  public isDragOver: boolean = false;
+
+  public constructor(private readonly cdr: ChangeDetectorRef) {}
+
+  public writeValue(value?: FileItem[]): void {
+    this.files.splice(0).push(...(value ?? []));
```

Comment:
We shouldn't be modifying input. I would recommend maintaining a separate internal property for this

Reply:
Actually, should `Files` be an input for this component? Users are adding Files through this component right?

Reply:
This is not an input property anymore.